cpp/src/io/orc/dict_enc.cu

```diff
@@ -205,47 +197,48 @@ CUDF_KERNEL void __launch_bounds__(block_size)
 {
   auto const col_idx    = blockIdx.x;
   auto const stripe_idx = blockIdx.y;
+  auto const t          = threadIdx.x;
   auto const& dict      = dictionaries[col_idx][stripe_idx];
   auto const& col       = columns[dict.column_idx];
 
   if (not dict.is_enabled) { return; }
 
-  auto const t         = threadIdx.x;
+  // Make a view of the hash map
+  auto const hash_fn        = hash_functor{col};
+  auto const equality_fn    = equality_functor{col};
+  using probing_scheme_type = cuco::linear_probing<map_cg_size, hash_functor>;
+
+  storage_ref_type const storage_ref{dict.map_slots.size(), dict.map_slots.data()};
+  // Make a view of the hash map.
+  auto hash_map_ref = cuco::static_map_ref{cuco::empty_key{KEY_SENTINEL},
+                                           cuco::empty_value{VALUE_SENTINEL},
+                                           equality_fn,
+                                           probing_scheme_type{hash_fn},
+                                           cuco::thread_scope_block,
+                                           storage_ref};
+
+  // Create a map ref with `cuco::insert` operator
+  auto has_map_find_ref = hash_map_ref.with_operators(cuco::find);
```

Comment:
```suggestion
  auto has_map_find_ref = hash_map_ref.rebind_operators(cuco::find);
```
Similar to https://github.com/rapidsai/cudf/pull/17052, we are deprecating `with_*` utilities and you may want to use `rebind_*` instead.